### PR TITLE
Convert part of futures to `async/await`

### LINKED ISSUE #119: Convert futures to async/await
We have a lot of async code now, but are still using some futures, converting them all (or as much as possible) would make the code clearer.

### PR DESCRIPTION
**Pull request by silathdiir**
Tries to convert part of futures to `async/await`. Fixes #119.

Signed-off-by: Steven Gu <asongala@163.com>

### CODE REVIEW

File: src/kv_client/mod.rs [hunk 1]
```diff
@@ -78,23 +78,27 @@ impl KvClient for KvRpcClient {
...
-    Compat01As03::new(fut.unwrap())
-        .map(|r| match r {
-            Err(e) => Err(ErrorKind::Grpc(e).into()),
-            Ok(r) => Ok(r),
-        })
-        .map(move |r| context.done(r))
+    let fut = match fut {
+        Err(e) => return context.done(Err(ErrorKind::Grpc(e).into())),
+        Ok(f) => f,
+    };
```
Comment (nrc) [inline src/kv_client/mod.rs]: Could this be written as `context.done(fut.and_then(async |f| Compat01As03::new(fut).await).map_err(|e| ErrorKind::Grpc(e).into()))` or something similar?

File: src/raw/client.rs [hunk 2]
```diff
@@ -256,7 +259,9 @@ impl Client {
                 requests::new_raw_scan_request(range, limit, self.key_only, self.cf.clone())
                     .execute(self.rpc.clone()),
             )
-        }
+        };
+
+        request.await
```
Comment (nrc) [inline src/raw/client.rs]: I think this function can be further simplified by directly returning the error in the first case and using `...execute.await` in the second.

File: src/raw/client.rs [hunk 3]
```diff
@@ -275,12 +280,12 @@ impl Client {
...
     /// ```
-    pub fn batch_scan(
+    pub async fn batch_scan(
         &self,
         ranges: impl IntoIterator<Item = impl Into<BoundRange>>,
         each_limit: u32,
-    ) -> impl Future<Output = Result<Vec<KvPair>>> {
-        if each_limit > MAX_RAW_KV_SCAN_LIMIT {
+    ) -> Result<Vec<KvPair>> {
+        let request = if each_limit > MAX_RAW_KV_SCAN_LIMIT {
```
Comment (nrc) [inline src/raw/client.rs]: Similarly to above

File: src/kv_client/mod.rs (hunk 1 above)
Comment (silathdiir) [reply src/kv_client/mod.rs]: @nrc It seems that the closure `async |f| Compat01As03::new(fut).await` could not return a Result (it returns a Future).

I tries to simplify the code as below for code review.

```
    let res = match fut {
        Ok(f) => Compat01As03::new(f).await,
        Err(e) => Err(e),
    };

    let context = tikv_stats(request_name);
    context.done(res.map_err(|e| ErrorKind::Grpc(e).into()))
```

**Comment by nrc** [comment]
@sticnarf PTAL

**Pull request merged**